transaction-decoder-tool/src/test/kotlin/linea/test/BlocksFetcher.kt

```diff
@@ -0,0 +1,77 @@
+package linea.test
+
+import build.linea.web3j.domain.toWeb3j
+import io.vertx.core.Vertx
+import linea.domain.Block
+import linea.web3j.toDomain
+import net.consensys.linea.BlockParameter.Companion.toBlockParameter
+import net.consensys.linea.async.AsyncRetryer
+import net.consensys.linea.async.toSafeFuture
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.web3j.protocol.Web3j
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import java.util.concurrent.atomic.AtomicLong
+import kotlin.time.Duration.Companion.milliseconds
+
+class BlocksFetcher(
+  val web3j: Web3j,
+  val vertx: Vertx = Vertx.vertx(),
+  val pollingChuckSize: UInt = 100U,
+  val log: Logger = LogManager.getLogger(BlocksFetcher::class.java)
+) {
+  fun fetchBlocks(
```

Comment:
This could be private I believe

Reply:
It was deliberately left public: it can be handy to just fetch some blocks and do/validate something ...

Reply:
But `consumeBlocks` allows exactly that